geoips/commandline/geoips_command.py

```diff
@@ -303,59 +331,192 @@ def _get_registry_by_interface_and_package(self, interface, package_name):
+                table_data += self._get_entry(product_dict, headers)
+        else:
+            table_data += self._get_entry(interface_registry, headers)
+        return table_data
+
+    def _get_entry(self, plugin_dict, headers):
+        """Retrieve the appropriate plugin entry given a list of valid headers.
+
+        Parameters
+        ----------
+        plugin_dict: dict
+            - The portion of the plugin_registry needed to access the appropriate plugin
+        headers: dict
+            - Dictionary of strings representing the key-value mapping of the
+              headers we'd like to output.
+
+        Returns
+        -------
+        plugin_entry: list of str
+            - A list of strings containing information about a plugin listed in the
+              order of headers.keys()
+        """
+        table_data = []
+        for plugin_key in list(plugin_dict.keys()):
+            plugin_entry = []
+            for header in list(headers.keys()):
+                if (
+                    header == "source_names"
+                    and plugin_dict[plugin_key]["interface"] != "products"
+                ):
+                    plugin_entry.append("N/A")
+                elif (
+                    header == "family"
+                    and plugin_dict[plugin_key]["interface"] == "products"
+                ):
+                    plugin_entry.append("N/A")
+                elif header == "plugin_name":
+                    plugin_entry.append(plugin_key)
+                else:
+                    plugin_entry.append(plugin_dict[plugin_key][header])
+            table_data.append(plugin_entry)
+        return table_data
+
+
+class GeoipsListCommon(GeoipsExecutableCommand):
+    """Class containing common optional arguments shared between list commands."""
+
+    command_name = "list_common"
+    subcommand_classes = []
+
+    def add_arguments(self):
+        """Add arguments to the list-subparser for the List Command."""
+        self.subcommand_parser.add_argument(
+            "--package_name",
+            "-p",
+            type=str,
+            default="all",
+            choices=self.plugin_package_names,
+            help="The GeoIPS package to list from.",
+        )
+        mutex_group = self.subcommand_parser.add_mutually_exclusive_group()
+        mutex_group.add_argument(
+            "--long",
+            "-l",
+            default=True,
+            action="store_true",
+            help="Flag representing the 'long' listing of a certain command.",
+        )
+        mutex_group.add_argument(
+            "--columns",
+            "-c",
+            type=str,
+            nargs="+",
+            default=None,
+            help="""Specific Headers of Data you'd like to see listed.
+                    For more in formation on headers available, run
+                    'geoips list <sub-cmd> <positional_args> --columns help'.
+                    """,
```

Comment:
When I call `geoips list interfaces --columns help` I get the following error. I don't think that `help` is currently being handled in `_get_headers_by_command`.

```
usage: To use, type `geoips list interfaces`. Optionally includes `-i` and `-p` flags for
seeing what has been implemented (-i) in a certain package (-p), or all packages.
For example, to see what interfaces have been implemented in a certain package,
run `geoips list interfaces -i -p <pkg_name>`, or you can remove the `-p` flag to
see what's been implemented in every package.
To use, type `geoips list <sub-cmd> <sub-sub-cmd>`. interfaces: error: Column header 'help', is not a valid header. Please select one or more of the following keys, which correspond to the appropriate value:
{'package': 'GeoIPS Package', 'plugin_type': 'Interface Type', 'interface': 'Interface Name', 'supported_families': 'Supported Families', 'docstring': 'Docstring', 'abspath': 'Absolute Path'}
```

Reply:
Okay, we should talk about what the appropriate functionality is here. Maybe we should have a separate flag for `--list-columns` rather than offering `--columns help`?

Reply:
I'd like to create an issue to discuss and fix the handling of columns. For now, in the interest of getting this merged today, I'm going to close this issue and just send a link to this convesration to @evrose54 so that one of us can create an issue on Monday.